src/libs/assay_type.py

```diff
@@ -103,6 +150,21 @@ def iter_names(cls, primary: BoolOrNone = None) -> Iterable[str]:
              False: iterate only over the names of derived assays, that is,
                     those for which a dataset of the given type has at least
                     one parent which is also a dataset.
+
+        >>> all_names = [elt for elt in AssayType.iter_names()]
+        >>> primary_names = [elt for elt in AssayType.iter_names(primary=True)]
+        >>> nonprimary_names = [elt for elt in AssayType.iter_names(primary=False)]
+        >>> all(elt in all_names for elt in primary_names)
+        True
+        >>> any(elt in nonprimary_names for elt in primary_names)
+        False
+        >>> all(elt in all_names for elt in nonprimary_names)
+        True
+        >>> any(elt in primary_names for elt in nonprimary_names)
+        False
+        >>> len(all_names) == len(primary_names) + len(nonprimary_names)
+        True
```

Comment:
```suggestion
        >>> assert all(elt in all_names for elt in primary_names)
        >>> assert not any(elt in nonprimary_names for elt in primary_names)
        >>> assert all(elt in all_names for elt in nonprimary_names)
        >>> assert not any(elt in primary_names for elt in nonprimary_names)
        >>> assert len(all_names) == len(primary_names) + len(nonprimary_names)
```
Your call.

Reply:
Ooh, I didn't know you could put an assert in a doctest.  Yes, this is better.